x-pack/legacy/plugins/lens/public/metric_visualization_plugin/auto_scale.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+interface Props {
+  children: React.ReactNode | React.ReactNode[];
+}
+
+interface State {
+  scale: number;
+}
+
+export class AutoScale extends React.Component<Props, State> {
+  private child: Element | null = null;
+  private parent: Element | null = null;
+
+  constructor(props: Props) {
+    super(props);
+
+    // An initial scale of 0 means we always redraw
+    // at least once, which is sub-optimal, but it
+    // prevents an annoying flicker.
+    this.state = { scale: 0 };
+  }
+
+  setParent = (el: Element | null) => {
+    if (this.parent !== el) {
+      this.parent = el;
+      setTimeout(() => this.scale());
+    }
+  };
+
+  setChild = (el: Element | null) => {
+    if (this.child !== el) {
+      this.child = el;
+      setTimeout(() => this.scale());
+    }
+  };
+
+  scale() {
+    const scale = computeScale(this.parent, this.child);
+
+    // Prevent an infinite render loop
+    if (this.state.scale !== scale) {
+      this.setState({ scale });
+    }
+  }
+
+  render() {
+    const { children } = this.props;
+    const { scale } = this.state;
+
+    return (
+      <div
+        className="autoscale-parent"
+        ref={this.setParent}
+        style={{
+          display: 'flex',
+          justifyContent: 'center',
+          position: 'relative',
+          maxWidth: '100%',
```

Comment:
The autosizing doesn't work if the height is too small because the child is allowed to flow out of the parent - `maxHeight: 100%` fixes this.

Reply:
It also needs `alignItems: 'center'` to also work correctly in the limited height scenario